libsolidity/parsing/Token.cpp

```diff
@@ -112,6 +119,10 @@ unsigned Token::extractM(string const& _literal)
 	{
 		return 0;
 	}
+	catch (invalid_argument& e)
+	{
+		return 0;
```

Comment:
As `0` is actually a valid unsigned (in e.g. `fixed0x8`), we should not return 0 on failure, we should return something like `unsigned(-1)`. Can you also create a test where a number that does not fit 64 bytes is used?